### PR TITLE
Fix mustSatisfyAnyOf logic (fix #101).

### LINKED ISSUE #101: `mustSatisfyAnyOf` fails the expected logic
## Summary

The function is admittedly completely undocumented, but from its type signature

    mustSatisfyAnyOf :: forall i o. [TxConstraints i o] -> TxConstraints i o

and the name, I would expect that it's meant to be equivalent to `any`, meaning if any of the items in the argument list is satisfied, the result will be satisfied as well. In a bit of pseudo-notation, 

    satisfied (mustSatisfyAnyOf xs) == any satisfied xs

As a reminder, the `TxConstraints` type is basically a conjunction of a list of `TxConstraint`, so it acts as an `all`; its `mempty` implementation is an empty list and is always trivially satisfied. So we can expand this to

    satisfied (mustSatisfyAnyOf xs) == any satisfied xs == any (all satisfied) xs

 That's not what happens. The actual function implementation is

    mustSatisfyAnyOf = singleton . MustSatisfyAnyOf . concatMap txConstraints

 The first thing the function does is to take a list of the conjunction lists flatten the list of `[[TxConstraint]]`, so we end up with

    satisfied (mustSatisfyAnyOf xs) == any (any satisfied) xs

## Steps to reproduce

    mustSatisfyAnyOf [mempty]

always fails.

## Expected behavior

    mustSatisfyAnyOf [mempty]

should be equal to `mempty`, i.e. it should be trivial to satisfy.

## System info (please complete the following information):

- OS: Ubuntu
- Version 21.10
- 58c093a49eb7a369865e361179d649264fc817a4

### PR DESCRIPTION
**Pull request by ak3n**
should fix #101

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x]  Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-ledger-constraints/src/Ledger/Constraints/OnChain.hs [hunk 1]
```diff
@@ -107,7 +108,7 @@ checkTxConstraint ctx@ScriptContext{scriptContextTxInfo} = \case
         $ V.findDatum dvh scriptContextTxInfo == Just dv
     MustSatisfyAnyOf xs ->
         traceIfFalse "Ld" -- "MustSatisfyAnyOf"
-        $ any (checkTxConstraint ctx) xs
+        $ or $ map (all (checkTxConstraint ctx)) xs
```
Comment (sjoerdvisscher) [inline plutus-ledger-constraints/src/Ledger/Constraints/OnChain.hs:111]: That's just `any (all (checkTxConstraint ctx)) xs` isn't it?

**Review by sjoerdvisscher — APPROVED** [review]
👍 Would be nice to use `any` in the implementation of `MustSatisfyAnyOf` though 😄

File: plutus-ledger-constraints/src/Ledger/Constraints/OnChain.hs (hunk 1 above)
Comment (ak3n) [reply plutus-ledger-constraints/src/Ledger/Constraints/OnChain.hs]: Ahah, yeah, missed that, thanks!

**Comment by silky** [comment]
Seems fine @ak3n ; but I note that the PIR gets larger, and some of the golden outputs (amounts) change? Is that _because_ the PIR got larger? Is that expected?

**Comment by ak3n** [comment]
@silky I think they get changed each time in `renderGuess.txt`.

At least it was true for https://github.com/input-output-hk/plutus-apps/commit/63123ffb527a37ba357e4cb38690625f945710b9, and for others commits in the history of that file.

**Review by silky — APPROVED** [review]

**Comment by silky** [comment]
@ak3n I'm happy then if you're happy :)

**Pull request merged**